### PR TITLE
Add events endpoint to be able to batch them

### PR DESCRIPTION
**Pull request by majst01**
Actually we create 2 phone home events per machine from metal-core, one per leaf, which creates a lot of api calls in the metal-api which raises cpu usage. Additionally the ingress treats this sometimes as a DDOS attack and sent GOAWAY back to the caller if a certain amount of request from the same client per time are fired.

### CODE REVIEW

**Review by mwennrich — APPROVED** [review]

File: cmd/metal-api/internal/service/machine-service.go [hunk 1]
```diff
@@ -328,6 +328,16 @@ func (r machineResource) webService() *restful.WebService {
 		Returns(http.StatusOK, "OK", v1.MachineRecentProvisioningEvents{}).
 		DefaultReturns("Error", httperrors.HTTPErrorResponse{}))
 
+	ws.Route(ws.POST("/events").
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/machine-service.go:331]: Up to this point we have no plural in our API namings, I think it's more consistent to use `/event`.

File: cmd/metal-api/internal/service/machine-service.go [hunk 2]
```diff
@@ -328,6 +328,16 @@ func (r machineResource) webService() *restful.WebService {
...
+		To(editor(r.addProvisioningEvents)).
+		Operation("addProvisioningEvents").
+		Doc("adds machine provisioning events").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(v1.MachineProvisioningEvents{}).
+		Writes(v1.MachineRecentProvisioningEventsMap{}).
+		Returns(http.StatusOK, "OK", v1.MachineRecentProvisioningEventsMap{}).
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/machine-service.go:338]: If there are many, is it really worth it to send them back?

File: cmd/metal-api/internal/service/machine-service.go [hunk 3]
```diff
@@ -2046,52 +2056,77 @@ func (r machineResource) getProvisioningEventContainer(request *restful.Request,
...
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	result := v1.MachineRecentProvisioningEventsMap{}
+	for machineID, event := range requestPayload {
+		ec, err := r.addProvisionEventForMachine(machineID, event)
+		if err != nil {
+			zapup.MustRootLogger().Error("unable to add provisioning event for machine", zap.String("machine", machineID), zap.Error(err))
+			continue
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/machine-service.go:2071]: Maybe the consumer should get an error back when machine events were not added?

File: cmd/metal-api/internal/service/machine-service_test.go [hunk 4]
```diff
@@ -663,6 +663,45 @@ func TestAddProvisioningEvent(t *testing.T) {
...
+	resp := w.Result()
+	defer resp.Body.Close()
+	require.Equal(t, http.StatusOK, resp.StatusCode, w.Body.String())
+	var result v1.MachineRecentProvisioningEventsMap
+	err = json.NewDecoder(resp.Body).Decode(&result)
+
+	require.NoError(t, err)
+	require.Equal(t, "0", result[machineID].IncompleteProvisioningCycles)
+	require.Len(t, result[machineID].Events, 1)
+	if len(result[machineID].Events) > 0 {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/machine-service_test.go:699]: This is ensured by the `require` in the line before.

File: cmd/metal-api/internal/service/machine-service.go (hunk 1 above)
Comment (majst01) [reply cmd/metal-api/internal/service/machine-service.go]: Done

File: cmd/metal-api/internal/service/machine-service.go (hunk 2 above)
Comment (majst01) [reply cmd/metal-api/internal/service/machine-service.go]: Changed to a proper response

File: cmd/metal-api/internal/service/machine-service.go (hunk 3 above)
Comment (majst01) [reply cmd/metal-api/internal/service/machine-service.go:2074]: still continue to preserve working events, but return if at least one error occurs now

File: cmd/metal-api/internal/service/machine-service_test.go (hunk 4 above)
Comment (majst01) [reply cmd/metal-api/internal/service/machine-service_test.go]: thanks, done

**Pull request merged**